### PR TITLE
34 - add emergency info

### LINKED ISSUE #34: [Passport user] - Edit emergency information page

### PR DESCRIPTION
**Pull request by brindasasi**
Resolves #34.

### Bonus
- [Fixes share-my-passport href](https://github.com/GCTC-NTGC/gc-accessibility-passport/pull/102/commits/160b9cfc2002c5830505837cad1ece7e2bca581b)
- [Fixes manager-info onSubmit route](https://github.com/GCTC-NTGC/gc-accessibility-passport/pull/102/commits/8000db150e7c1c00cf94e897554791b20a71cbd7)

### CODE REVIEW

File: pages/emergency-info.tsx [hunk 1]
```diff
@@ -0,0 +1,307 @@
+import { PaperClipIcon } from "@heroicons/react/solid";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React from "react";
+import { FormProvider, SubmitHandler, useForm } from "react-hook-form";
+import { useIntl } from "react-intl";
+import { colorMap } from "../components/Button";
+import { Input, Submit, TextArea } from "../components/formComponents";
```
Comment (mnigh) [inline pages/emergency-info.tsx]:
```suggestion
import { Input, TextArea } from "../components/formComponents";
```

File: pages/emergency-info.tsx [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+  const intl = useIntl();
+  const { push } = useRouter();
+  const methods = useForm<FormValues>();
+  const { handleSubmit } = methods;
+  const onSubmit: SubmitHandler<FormValues> = async () => {
+    push("/passport"); // TODO: should return to manager dashboard if user is manager.
+  };
+
+  return (
+    <Layout
```
Comment (mnigh) [inline pages/emergency-info.tsx]: Based on [design](https://www.figma.com/file/SqRfWwcyuclFvGCq47zJf6/Passport---V5.0.0?node-id=2%3A11), missing breadcrumbs: _My passport / Emergency Information_

File: pages/emergency-info.tsx [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+                  })}
+                </p>
+                <div
+                  data-h2-border="b(black, all, solid, s)"
+                  data-h2-margin="b(top, s)"
+                  data-h2-padding="b(all, m)"
+                  data-h2-display="b(flex)"
+                >
+                  <Link href={"#"}>
+                    <a data-h2-margin="b(right, m)">{"Upload"}</a>
```
Comment (mnigh) [inline pages/emergency-info.tsx]: Is this meant to be `<input type="file" />`?

File: pages/emergency-info.tsx [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+                  </Link>
+                </div>
+
+                <div
```
Comment (mnigh) [inline pages/emergency-info.tsx]: Consider replacing these two file rows with for alignment and responsiveness:

```
                <div
                  data-h2-margin="b(top-bottom, s)"
                  data-h2-display="b(block) s(flex)"
                >
                  <span data-h2-margin="b(right, xs)">
                    <PaperClipIcon style={{ width: "1.2rem" }}> </PaperClipIcon>
                  </span>
                  <Link href="#">
                    <a data-h2-margin="b(right, xs)">
                      {intl.formatMessage({
                        defaultMessage: "my_paramedical_needs.pdf(3MB)",
                      })}
                    </a>
                  </Link>
                  <Link href={"#"}>
                    <a
                      data-h2-margin="b(left, auto)"
                      data-h2-display="b(block)"
                    >
                      Remove
                    </a>
                  </Link>
                </div>
                <div
                  data-h2-margin="b(top-bottom, s)"
                  data-h2-display="b(block) s(flex)"
                >
                  <span data-h2-margin="b(right, xs)">
                    <PaperClipIcon style={{ width: "1.2rem" }}> </PaperClipIcon>
                  </span>
                  <Link href="#">
                    <a data-h2-margin="b(right, xs)">
                      {intl.formatMessage({
                        defaultMessage: "my_evacuation_plan.pdf(3MB)",
                      })}
                    </a>
                  </Link>
                  <Link href={"#"}>
                    <a
                      data-h2-margin="b(left, auto)"
                      data-h2-display="b(block)"
                    >
                      Remove
                    </a>
                  </Link>
                </div>
```

File: pages/emergency-info.tsx [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+import Layout from "../components/Layout";
+import Page, { LeftSection, RightSection } from "../components/Page";
+import Button from "../components/Button";
+
+type FormValues = {
+  currentEmail: string;
+  newEmail: string;
+  password: string;
+};
+const bold = (msg: string): React.ReactNode => (
```
Comment (mnigh) [inline pages/emergency-info.tsx]: Consider changing the name of this to `strong` and using <strong> tags instead of css within the function itself.

**Review by mnigh — CHANGES_REQUESTED** [review]
Good start. Please see my suggestions and comments.

File: pages/passport/emergency-info.tsx [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+      })}
+      data-h2-width="b(100) l(75)"
+      data-h2-padding="b(all, none)"
+      crumbs={[
+        {
+          title: intl.formatMessage({
+            defaultMessage: "My passport",
+            description: "Breadcrumb title.",
+          }),
+        },
```
Comment (yonikid15) [inline pages/passport/emergency-info.tsx]:
```suggestion
      crumbs={[
        {
          title: intl.formatMessage({
            defaultMessage: "My passport",
            description: "Breadcrumb title.",
          }),
          href: "/passport",
        },
```

File: pages/passport/emergency-info.tsx [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+import Layout from "../../components/Layout";
+import Page, { LeftSection, RightSection } from "../../components/Page";
+import Button from "../../components/Button";
+import { strong } from "../../helpers/format";
+
+type FormValues = {
+  currentEmail: string;
+  newEmail: string;
+  password: string;
+};
```
Comment (yonikid15) [inline pages/passport/emergency-info.tsx]:
```suggestion
type FormValues = {
  description: string;
  firstName: string;
  lastName: string;
  telephone: string;
};
```

File: pages/passport/emergency-info.tsx [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+            description: "Breadcrumb title.",
+          }),
+        },
+      ]}
+    >
+      <Page>
+        <LeftSection>
+          <FormProvider {...methods}>
+            <form onSubmit={handleSubmit(onSubmit)}>
+              <p>
```
Comment (yonikid15) [inline pages/passport/emergency-info.tsx]:
```suggestion
              <p data-h2-margin="b(top, none)">
```

File: pages/passport/emergency-info.tsx [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+        <RightSection>
+          <div data-h2-margin="b(top-bottom, l)">
+            <h4 data-h2-font-size="b(h4)" data-h2-margin="b(bottom, none)">
+              {intl.formatMessage(
+                {
+                  defaultMessage: "Permissions",
+                },
+                { strong },
+              )}
+            </h4>
```
Comment (yonikid15) [inline pages/passport/emergency-info.tsx]:
```suggestion
          <div>
            <h4 data-h2-font-size="b(h4)" data-h2-margin="b(top-bottom, none)">
              {intl.formatMessage({
                defaultMessage: "Permissions",
              })}
            </h4>
```

**Review by yonikid15 — CHANGES_REQUESTED** [review]
Left some minor suggested changes, but overall looks good!

**Review by yonikid15 — APPROVED** [review]
Love it

**Pull request merged**